### ISSUE TITLE
cuda inverse issue

### ISSUE THREAD

**Issue by xiao7199** [pytorch/pytorch#3585]
The execution time of CUDA inverse is weird.  The first inverse operation will takes longer time than the consecutive inverse operations.
Here is the code:

```python
data = Variable(torch.rand(256,256).type(torch.FloatTensor)).cuda()
a = time.time()
torch.inverse(data)
b = time.time()
torch.inverse(data)
c = time.time()
torch.inverse(data)
d = time.time()
print(d-c,c-b,b-a)
```

The result is
 `(0.006289958953857422, 0.006345033645629883, 0.2919008731842041)`
So the first inverse will always take significant longer time than the next several inverse operation. 
Any idea?

**Comment by apaszke** [comment]
You're not really benchmarking CUDA time - CPU and GPU run asychronously, so you need to synchronize them:
```python
import time                                                        
import torch                                                       
from torch.autograd import Variable                                
                                                                   
def get_time():                                                    
    torch.cuda.synchronize()                                       
    return time.time()                                             
                                                                   
                                                                   
data = Variable(torch.rand(256,256).type(torch.FloatTensor)).cuda()
a = get_time()                                                     
torch.inverse(data)                                                
b = get_time()                                                     
torch.inverse(data)                                                
c = get_time()                                                     
torch.inverse(data)                                                
d = get_time()                                                     
print(d-c,c-b,b-a)                                                 
```
Still, even in this version the first call is slower - but this is expected. `inverse` needs to allocate some working memory, the first call will allocate it, and it will be cached and reused by subsequent calls, making them a lot faster.

**Comment by xiao7199** [comment]
Thanks for your reply.

My algorithm needs 1 CNN forward and 1 256*256 matrix inverse.
Now the running time for CNN part is around 0.009s each image but the inverse operation takes about 0.70s which heavily drag down FPS.

My question is: 
Can I first initialized 'inverse' somewhere by allocating the working memory and then reuse the cached ready function within the image loop to accelerate the computation?
I have tried this way but still got the same running time, ~0.7s

Thanks
